### PR TITLE
Initial commit.

### PR DESCRIPTION
**Pull request by mookerji**
Includes:
- Some basic design notes.
- Basic testing, CI, and build packaging.
- SBP subscription from the broker.
- Basic logging.

/cc @mfine @JoshuaGross @fnoble @jacobmcnamee

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+all: sky-connect sky-connect-verbose
```
Comment (mfine) [inline Makefile]: Oh god, this was gross - sorry. Should use like sane use of `CFLAGS` and `LDFLAGS`.

Comment (mfine) [inline Makefile]: wait, is this makefile even being used?

**Comment by JoshuaGross** [comment]
It looks like `libsbp` isn't being used? and probably doesn't need to be? like we probably just want to stream directly from the device to skylark without encoding/decoding anything, right?

File: src/connect.c [hunk 2]
```diff
@@ -0,0 +1,54 @@
+/*
```
Comment (mfine) [inline src/connect.c]: Maybe just call this `skylark.c`?

File: Makefile (hunk 1 above)
Comment (mookerji) [inline Makefile]: Nope

File: src/serial.c [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  }
+  if (tcsetattr(fd, TCSANOW, &termattr) < 0)
+  {
+    log_error("tcsetattr(TCSANOW)");
+    return -DEVICE_CONF_ERROR;
+  }
+  return fd;
+}
+
+int set_serial_baud_rate(int fd, int baud)
```
Comment (mfine) [inline src/serial.c]: Might be preferrable to have `set_serial_baud_rate_linux` and `set_serial_baud_rate_apple` functions and do the switching around which one you call.

**Comment by mfine** [comment]
Why do you hate spaces inside C functions? Looking good...

**Comment by mookerji** [comment]
TODO: 
- [] Tests
- [] Fleshing out design ideas not yet done
- [] Threading

**Comment by mookerji** [comment]
@JoshuaGross re: libsbp, I had designs of reading device settings out, but perhaps that's a waste of time since we'd ultimately want to do something else anyway.

**Comment by mookerji** [comment]
I'll leave it out for the time being.

**Comment by mookerji** [comment]
Does anyone have any recommendations for unit testing a libcurl-based library in C?

*shudder*

**Comment by mfine** [comment]
I think we should punt reading out the device settings and pass in serial id or device uid as a parameter in these libraries. It's awesome to see nice simple code like this and in something like [sky-connect.py](https://github.com/swift-nav/skylark-tools/blob/master/python/sky-connect.py) that doesn't take an sbp dependency and doesn't have to do a handshake with the device and parse protocol in order to send bytes up to skylark. And keep in mind we may change the way we generate uid's going forward. Love keeping this library lightweight for now.

**Comment by JoshuaGross** [comment]
Huh, the unit testing question is interesting. I did some rudimentary digging and it looks like all the HTTP mocking libraries exist for higher-level languages: Java, Python, JS it's pretty easy - but it seems fairly difficult in C since you can't just take over a runtime. Seems like you'd need to do something custom here, maybe even open up a port and start a test server for unit tests? Since you'd be likely setting up a custom test rig anyway, I would suggest writing a really simple Node.js script that would open up some port, verify that something happens and then exit with an error/success code or time out after a couple seconds.

**Comment by mookerji** [comment]
I made several follow on issues to complete some work, but I think this is done for the time being.

**Pull request merged**